src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java

```diff
@@ -226,7 +231,57 @@ private void invokeDirectoryNio(PathOperation<? extends PathItem> pathOperation)
 				LogUtil.exception(Level.DEBUG, e, "Unexpected failure");
 			}
 		}
-    }
+	}
+
+	private boolean invokePathCreate(final PathOperation<? extends PathItem> pathOp) {
+		final String path = pathOp.dstPath();
+		StreamManager streamManager = StreamManager.create(URI.create(uriSchema));
+		final String scopeName = path.substring(0, path.indexOf("/"));
+		final String streamName = path.substring(path.indexOf("/")+1);
+
+		final Function<String, Map<String, String>> createScopeComputeIfAbsent = (final String k) -> {
+			streamManager.createScope(k);
+			return new ConcurrentHashMap<String, String>();
+		};
+
+		final Function<String, String> createStreamComputeIfAbsent = (final String k) -> {
+			streamManager.createStream(scopeName, k, streamConfig);
+			return k;
+		};
+
+		if (null != scopeMap.computeIfAbsent(scopeName,	createScopeComputeIfAbsent)) {
```

Comment:
this condition will be `true` always because the map (new either existing previously) will be returned

Reply:
Does it mean that then we generally need to throw out the condition and just call the function?

Reply:
not just call but save the result into the local variable in order to not invoke `scopeMap.get` after again